### PR TITLE
Package core_unix.v0.14.0

### PR DESCRIPTION
**Pull request by aalekseyev**
In Jane Street v0.15 we're about to split up core library and move most of it into package `core_unix`.

This PR adds a compatibility package `core_unix.v0.14.0` which should be largely compatible with the upcoming `core_unix.v0.15.0`, while being implemented on top of `core.v0.14.*`.

cc @kit-ty-kate 

Signed-off-by: Arseniy Alekseyev <aalekseyev@janestreet.com>

### CODE REVIEW

File: packages/core_unix/core_unix.v0.14.0/opam [hunk 1]
```diff
@@ -0,0 +1,29 @@
+opam-version: "2.0"
+version: "v0.14.0"
```
Comment (mseri) [inline packages/core_unix/core_unix.v0.14.0/opam:2]:
```suggestion
```

Comment (aalekseyev) [reply packages/core_unix/core_unix.v0.14.0/opam]: Thanks. I wasn't sure why or whether this field was needed, but I cargo-culted it from somewhere.

**Comment by mseri** [comment]
No worries. Name and version are inferred from the path in the opam repository, so we have started removing them when they are left over in the linting stage

**Comment by mseri** [comment]
Looks good, thanks

**Pull request merged**